### PR TITLE
[java-generator] Fix the download directory creation steps

### PR DESCRIPTION
**Pull request by andreaTP**
## Description

The target download folder is still slightly off 😓 
This should fix it, cc. @manusa 

## Type of change
 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: java-generator/core/src/main/java/io/fabric8/java/generator/URLJavaGenerator.java [hunk 1]
```diff
@@ -57,13 +57,13 @@ public void run(File outputDirectory) {
       throw new JavaGeneratorException(
           "Download directory is required");
     }
+    if (!downloadDirectory.exists() && !downloadDirectory.mkdirs()) {
+      throw new JavaGeneratorException("Unable to create download directory " + downloadDirectory.getAbsolutePath());
+    }
```
Comment (rohanKanojia) [inline java-generator/core/src/main/java/io/fabric8/java/generator/URLJavaGenerator.java:62]: Do you think `downloadDirectory.isDirectory()`(on line 63) check is necessary after moving this check-up? Won't `downloadDirectory.mkdirs()` returning `true` would guarantee this?

Comment (andreaTP) [reply java-generator/core/src/main/java/io/fabric8/java/generator/URLJavaGenerator.java:62]: yes, you are right, this is redundant, happy to remove the condition!

Comment (andreaTP) [reply java-generator/core/src/main/java/io/fabric8/java/generator/URLJavaGenerator.java:62]: reverted as we have a specific test looking for the folder colliding with a file, I think is fine to leave it as in my first proposal ...

**Review by manusa — APPROVED** [review]

**Comment by manusa** [comment]
Relates to #4710

**Comment by manusa** [comment]
Was this reported as a bug anywhere so we can link?

**Comment by andreaTP** [comment]
> Was this reported as a bug anywhere so we can link?

No I can open a tracker if needed

**Comment by manusa** [comment]
> > Was this reported as a bug anywhere so we can link?
> 
> No I can open a tracker if needed

No need. I recall we might have talked about something like this before, so wanted to make sure we linked the issue if it was there.

**Pull request merged**